helpers/spark-submit.py

```diff
@@ -24,16 +24,19 @@
         os.environ['SPARK_HOME'] = os.environ['SNAP']
 
     STATIC_DEFAULTS_CONF_FILE = utils.get_static_defaults_conf_file()
-    DYNAMIC_DEFAULTS_CONF_FILE = utils.get_dynamic_defaults_conf_file()
     ENV_DEFAULTS_CONF_FILE = utils.get_env_defaults_conf_file()
+    primary = utils.retrieve_primary_service_account_details()
+    primary_username = primary['spark.kubernetes.authenticate.driver.serviceAccountName']
+    primary_namespace = primary['spark.kubernetes.namespace']
 
     snap_static_defaults = utils.read_property_file(STATIC_DEFAULTS_CONF_FILE)
-    setup_dynamic_defaults = utils.read_property_file(DYNAMIC_DEFAULTS_CONF_FILE) if os.path.isfile(DYNAMIC_DEFAULTS_CONF_FILE) else dict()
+    setup_dynamic_defaults = utils.retrieve_primary_service_account_details()
```

Comment:
why are we calling retrieve_primary_service_account_details twice? line 33 and line 28?

To be honest for better readability of the code, I would really just compress this business logic into a single function 

`dynamic_defautls = utils.get_dynamic_defaults(service_accont)`

with

`def get_dynamic_defaults(service_account: Optional[str]) -> Dict`

I put this as optional since, following the ticket description, one could also provide a service account via CLI --service-account=namespace:service-account. If no service account is provided, we actually do the inference that you are doing here, otherwise we use the one provided by the user